google-cloud-storage/src/main/java/com/google/cloud/storage/RewindableContent.java

```diff
@@ -55,6 +59,10 @@ public final boolean retrySupported() {
     return false;
   }
 
+  abstract String getMd5() throws IOException;
+
+  abstract String getCrc32c() throws IOException;
```

Comment:
I'm not sure this is the best place for this logic to live. Especially in the case of the File based one, the entire file would have to be read from disk before processing could proceed. gRPC has streaming checksum support, and JSON should be getting it soon as well.